EncoreTickets.SDK/Authentication/AuthenticationService.cs

```diff
@@ -0,0 +1,64 @@
+﻿using EncoreTickets.SDK.v1;
+using RestSharp;
+
+namespace EncoreTickets.SDK.Authentication
+{
+    /// <inheritdoc />
+    /// <summary>
+    /// The authentication service.
+    /// </summary>
+    public class AuthenticationService : BaseCapabilityServiceApi
+    {
+        private readonly string endpoint;
+
+        /// <summary>
+        /// Initializes an instance for the authentication service.
+        /// </summary>
+        /// <param name="context">The API context.</param>
+        /// <param name="host">The service host.</param>
+        /// <param name="loginEndpoint">The endpoint for login method.</param>
+        public AuthenticationService(ApiContext context, string host, string loginEndpoint)
+            : base(context, host)
+        {
+            endpoint = loginEndpoint;
+        }
+
+        /// <summary>
+        /// Get an API context with data set for an authenticated user.
+        /// </summary>
+        /// <returns>The API context</returns>
+        public ApiContext Authenticate()
+        {
+            switch (context.AuthenticationMethod)
```

Comment:
A note that can probably be fixed as part of final PR with refactoring changes. Every time you add a new authentication method, this switch statement will grow as well as the one in the next method. Also, you'll add a separate method for authenticating users like the ```JwtLogin()``` below. Maybe it's better to introduce a class hierarchy for ```AuthenticationService``` and select the appropriate concrete implementation via a factory?

Reply:
the right way to do it is to have auth interface with authenticate method which returns a context and have multiple implementations for each type and have a separate factory (static method of class) to determine which class to return according to the type. But, I don't think we will go beyond JWT so I think this is good enough

Reply:
Yes, I thought about it, but at the moment this service looks very simple and constant, so I believe that now additional complexity with the fabric is not required